src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -10,28 +12,71 @@
      * @param userInput full user input string
      * @return the command based on the user input
      */
+    public static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>^[\\S]+)(?<arguments>[\\d\\s\\S]*$)");
     public Command parseCommand(String userInput) {
 
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
+        final String commandWord = matcher.group("commandWord").trim();
+        final String arguments = matcher.group("arguments").trim();
+
         switch (commandWord) {
 
         case AddCommand.COMMAND_WORD:
+
             return prepareAdd(arguments);
 
-        case DeleteCommand.COMMAND_WORD:
-            return prepareDelete(arguments);
+        default:
+            return new HelpCommand();
+        }
+    }
+
+    private Command prepareAdd(String arguments) {
 
-        case ClearCommand.COMMAND_WORD:
-            return new ClearCommand();
+        String[] args = splitArgsForAddItem(arguments);
+        String description;
+        String prices;
+        description = args[1];
+        prices = args[2];
+        double price = Double.parseDouble(prices);
+        return new AddCommand(description,price);
+    }
 
-        case ListCommand.COMMAND_WORD:
-            return new ListCommand();
+    private String[] splitArgsForAddItem(String arguments) throws NullPointerException{
```

Comment:
Maybe you could rename to splitArgsForAddCommand

Reply:
Noted! Thanks ^^